lib/StaticNarrative/narrative_ref.py

```diff
@@ -8,8 +8,12 @@
 
 
 class NarrativeRef:
-    def _less_than_zero(self: "NarrativeRef", number: str | int, name: str) -> int:
+    """Lightweight class representing a KBase reference."""
+
+    def _less_than_zero(self: "NarrativeRef", number: str | int | None, name: str) -> int:
         err_msg = f"The Narrative {name} must be an integer > 0, not {number}"
+        if number is None or not str(number).isdigit():
```

Comment:
Casting to int meant that if you put in values like `4.5`, they would get converted to `4` and the ref would appear valid, so I made this a bit more strict.